---
title: "cil-poly-var haplotypes"
output: html_document
date: "2025-01-13"
---

libraries
```{r}
library(tidyverse)
library(ape)
library(pegas)
```


```{r}
data <- read.dna("/home/kimberly.ledger/rockfish_mb/data/cpv_fasta/cpv_minidloop_aligned.fasta", format = "fasta")
print(data)
```

compute haplotypes 
```{r}
haps <- haplotype(data)
haps
```

```{r}
#get info about what samples have which haplotypes
hapInfo <- stack(setNames(attr(haps,"index"),rownames(haps)))
names(hapInfo) <- c("index","haplotype")
head(hapInfo)

#need to join the sample names 
meta_aligned <- read.csv("/home/kimberly.ledger/rockfish_mb/data/cpv_fasta/cpv_meta.csv") %>%
  mutate(Name = ifelse(Name == '31082_consensus extraction', '31082_consensus_extraction', Name))

merged <- data.frame(cbind(hapInfo,meta_aligned[hapInfo$index,]))
head(merged)
```


color by species 
```{r}
pie <- table(merged$haplotype, merged$Organism)
head(pie)
```

```{r}
net <- haploNet(haps)

# Define custom colors (modify as needed)
my_colors <- c("green", "purple", "orange")

plot(net,size=attr(net,"freq"),pie=pie, bg = my_colors)
legend("bottomleft", colnames(pie), col=my_colors, pch=19, ncol=2)
```


is it possible to filter this by haplotype frequency? 
```{r}
hap_freq <- as.data.frame(summary(haps))
colnames(hap_freq) <- c("frequency")

haps_to_keep <- hap_freq %>%
  filter(frequency > 2)
haps_to_keep$haplotype <- rownames(haps_to_keep)
```


join the metadata and filter 
```{r}
samples_to_keep <- merged %>%
  filter(haplotype %in% haps_to_keep$haplotype)

data_filtered <- data[rownames(data) %in% samples_to_keep$Name, ]
data_filtered
```

compute haplotypes 
```{r}
haps_filtered <- haplotype(data_filtered)
haps_filtered
```


get the metadata 
```{r}
#get info about what samples have which haplotypes
hapInfo_filtered <- stack(setNames(attr(haps_filtered,"index"),rownames(haps_filtered)))
names(hapInfo_filtered) <- c("index","haplotype")
head(hapInfo_filtered)

meta_filtered <- meta_aligned %>%
  filter(Name %in% samples_to_keep$Name)

merged_filtered <- data.frame(cbind(hapInfo_filtered, meta_filtered[hapInfo_filtered$index,]))
head(merged_filtered)
```


color by species 
```{r}
pie_filtered <- table(merged_filtered$haplotype, merged_filtered$Organism)
head(pie_filtered)
```

```{r}
net_filtered <- haploNet(haps_filtered)

# Define custom colors (modify as needed)
my_colors <- c("#7f7f7f", "#77ab31", "#2bbab8")

#png("/home/kimberly.ledger/rockfish_mb/figures/haplotypes.png", width = 2400, height = 1800, res = 300)  # Set dimensions & resolution

plot(net_filtered,size=attr(net_filtered,"freq"),pie=pie_filtered, bg = my_colors, labels = rep("", length(net)))
legend("bottomright", colnames(pie_filtered), col=my_colors, pch=19, ncol=2)

#dev.off()
```

```{r}
png("/home/kimberly.ledger/rockfish_mb/figures/haplotypes_legend.png", width = 2400, height = 1800, res = 300)  # Set dimensions & resolution
plot(net_filtered,size=attr(net_filtered,"freq"),pie=pie_filtered, bg = my_colors)
legend("topleft", colnames(pie_filtered), col=my_colors, pch=19, ncol=2)
dev.off()
```


name each haplotype by it's composition 

```{r}
haplotype_freqs <- data.frame(pie_filtered) %>%
  group_by(Var1) %>% 
  mutate(total= sum(Freq)) %>%
  mutate(proportion = Freq/total) #%>%
  #ungroup() %>%
  #pivot_wider(names_from = Var2, values_from = Freq) %>%
  #dplyr::rename(haplotype = Var1)

haplotype_freqs_table <- data.frame(pie_filtered) %>%
  pivot_wider(names_from = Var2, values_from = Freq) %>%
  dplyr::rename(haplotype = Var1)

sum(haplotype_freqs_table$`Sebastes ciliatus`)
sum(haplotype_freqs_table$`Sebastes variabilis`)
```


#### not used in manuscript: 
load matrix of mini-loop
```{r}
matrix <- read.csv("/home/kimberly.ledger/rockfish_mb/data/cil-poly-var-dloop-matrix.csv")
```

long format 
```{r}
matrix_long <- matrix %>%
  pivot_longer(cols = c(2:202), names_to = "sample2", values_to = "identity") %>%
  rename(sample1 = X) %>%
  filter(!is.na(identity)) %>%
   mutate(sample2 = sub("^X", "", sample2))
```

load sample metadata
```{r}
meta <- read.csv("/home/kimberly.ledger/rockfish_mb/data/cil-poly-var-dloop-meta.csv") %>%
  filter(Name != "Nucleotide alignment") %>%
  filter(Name != "Nucleotide alignment consensus tree") %>%
  select(!Description) 
```

join
```{r}
matrix_long <- matrix_long %>%
  left_join(meta, by = c("sample1" = "Name")) %>%
  rename(organism1 = Organism) %>%
  left_join(meta, by = c("sample2" = "Name")) %>%
  rename(organism2 = Organism) 
```


first look closer at ciliatus
```{r}
matrix_long %>%
  filter(sample1 == "33267_consensus") %>%
  filter(organism1 == organism2) %>%
  group_by(identity) %>%
  summarize(count = n()) %>%
  arrange(desc(identity))

matrix_long %>%
  filter(sample1 == "33267_consensus") %>%
  filter(organism1 != organism2) %>%
  group_by(identity) %>%
  summarize(count = n()) %>%
  arrange(desc(identity))
```


how about polyspinus
```{r}
matrix_long %>%
  filter(sample1 == "ABLG9548_consensus") %>%
  filter(organism1 == organism2) %>%
  group_by(identity) %>%
  summarize(count = n()) %>%
  arrange(desc(identity))

matrix_long %>%
  filter(sample1 == "ABLG9548_consensus") %>%
  filter(organism1 != organism2) %>%
  group_by(identity) %>%
  summarize(count = n()) %>%
  arrange(desc(identity))
```

