---
title: "rockfish mb - testing abl and burke museum tissues"
author: "Kimberly Ledger"
date: "2024-08-14"
output: html_document
---

NOTE TO SELF - remove ABLGs from test that ended up going into reference DB!


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r load-libraries}
library(tidyverse)
```

load taxonomic assignment of asv's
```{r}
## untrimmed
tax <- read.csv("/home/kimberly.ledger/rockfish_mb/data/taxonomy_custom536_20250103.csv") %>%
  select(!X) %>%
  rename(ASV = qseqid)
## trimmed 
tax <- read.csv("/home/kimberly.ledger/rockfish_mb/data/taxonomy_custom536_20250103_trimmed.csv") %>%
  select(!X) %>%
  rename(ASV = qseqid)
```

load asv table
```{r}
## untrimmed 
asv_table <- read.csv("/genetics/edna/workdir/sebastes/tissues/untrimmed_20241231/filtered/outputs/ASVtable.csv")

## trimmed
asv_table <- read.csv("/genetics/edna/workdir/sebastes/tissues/trimmed_20241204/filtered/outputs/ASVtable.csv")
```

convert asv table to taxon table
```{r}
last <- ncol(asv_table)
first <- which(startsWith(names(asv_table), "ASV"))[1]

taxon_table <- asv_table %>%
  pivot_longer(cols = c(first:last), names_to = "ASV", values_to = "reads") %>%
  left_join(tax, by="ASV") %>%
  group_by(X, taxon) %>%
  summarize(tot_reads = sum(reads)) %>%
  filter(tot_reads > 0) %>%
  rename(sample_id = X)
```

```{r}
asv_table %>%
  pivot_longer(cols = c(first:last), names_to = "ASV", values_to = "reads") %>%
  left_join(tax, by="ASV")%>%
  filter(X %in% c("43643", "43665", "43666")) %>%
  filter(reads > 0)
```




sequencing read numbers were pretty low per sample, but will still apply a small threshold here to remove very low read number samples 
```{r}
reads_per_sample <- taxon_table %>%
  group_by(sample_id) %>%
  mutate(reads_per_sample = sum(tot_reads))

keepers <- reads_per_sample %>%
  filter(reads_per_sample > 100)

taxon_table_filtered <- taxon_table %>%
  filter(sample_id %in% keepers$sample_id) %>%
  filter(!sample_id %in% c("34137", "34138", "34139", "34140")) ##, "43649"))
```


read in file that has the tissue ids for the rockfish samples 
```{r}
ablg <- read.csv("~/rockfish_mb/data/ABLG_BURKE_rockfishDNA.csv") %>%
  filter(common_name != "yelloweye") %>%  ## need to pick new tissue samples to retest at some point
  filter(sample_id != "ABLG11655") #%>% ## misID of tissue
  #filter(sample_id != "43649") ## remove one remaining diploproa (splitnose)
```

how many species tested? 
```{r}
ablg %>% select(common_name) %>% unique()
```
```{r}
ablg %>% group_by(common_name) %>%
  summarize(count =n ()) %>%
  summarize(min_count = min(count),
            mean_count = mean(count),
            max_count = max(count))
```



```{r}
rockfish <- taxon_table_filtered %>%
  left_join(ablg, by = "sample_id") %>%
  group_by(sample_id) %>%
  mutate(reads_per_sample = sum(tot_reads)) %>%
  mutate(prop_reads = tot_reads/reads_per_sample) %>%
  mutate(correct_id = if_else(taxon == species_name, "yes", "no")) %>%
  mutate(correct_id = replace_na(correct_id, "no")) %>%
  mutate(species_name_abbr = sub("^(\\w)\\w*\\s(\\w+)", "\\1. \\2", species_name))
```

remove the ABLG's that were used in the reference db from this validation test
```{r}
ablg_id <- read.csv("/home/kimberly.ledger/rockfish_mb/data/rockfish_reference_db_536.csv") %>%
  rename(accession_id = Name) %>%
  filter(grepl("ABLG", accession_id))
```

need to change the correct ID column to account for the genus pair IDs that aren't wrong, they just don't get to species-level

```{r}
rockfish <- rockfish %>%
  mutate(correct_id = ifelse(taxon == "Sebastes rc" & species_name == "Sebastes crameri" | species_name == "Sebastes reedi", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes cv" & species_name == "Sebastes ciliatus" | species_name == "Sebastes variabilis", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes vw" & species_name == "Sebastes variegatus" | species_name == "Sebastes wilsoni", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes ev" & species_name == "Sebastes emphaeus" | species_name == "Sebastes variegatus", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes em" & species_name == "Sebastes entomelas" | species_name == "Sebastes mystinus", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes cp" & species_name == "Sebastes ciliatus" | species_name == "Sebastes polyspinis", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes de" & species_name == "Sebastes diaconus" | species_name == "Sebastes entomelas", "yes", correct_id),
         correct_id = ifelse(taxon == "Sebastes cr" & species_name == "Sebastes chlorostictus" | species_name == "Sebastes rosenblatti", "yes", correct_id))
```

make a plot 
```{r}
p1 <- rockfish %>%
  filter(species_name != "Sebastolobus altivelis") %>%
  filter(species_name != "Sebastolobus macrochir") %>%  ## not going to worry about sebastolobus 
  filter(!sample_id %in% ablg_id$accession_id) %>%
  ggplot(aes(x=sample_id, y=prop_reads, fill=correct_id)) +
  geom_bar(stat = "identity") + 
  theme_bw() +
  facet_wrap(~species_name_abbr, scales = "free_x", nrow = 3) + 
  labs(
    y = "relative read abundance (%)",
    x = "sample_id") + 
  theme(axis.text.x = element_blank(),
    #axis.text.x = element_text(angle = 90, hjust = 0.95),
    legend.text = element_text(size = 8),
    legend.key.size = unit(0.3, "cm"),
    legend.position = "bottom",
    legend.title = element_blank()
  )

p1 
```

```{r}
p2 <- rockfish %>%
  filter(species_name != "Sebastolobus altivelis") %>%
  filter(species_name != "Sebastolobus macrochir") %>%  ## not going to worry about sebastolobus 
  filter(!sample_id %in% ablg_id$accession_id) %>%
  ggplot(aes(x=sample_id, y=prop_reads, fill=taxon)) +
  geom_bar(stat = "identity") + 
  theme_bw() +
  facet_wrap(~species_name_abbr, scales = "free_x", nrow = 3) + 
  labs(
    y = "relative read abundance (%)",
    x = "sample_id") + 
  theme(axis.text.x = element_blank(),
    #axis.text.x = element_text(angle = 90, hjust = 0.95),
    legend.text = element_text(size = 8),
    legend.key.size = unit(0.3, "cm"),
    legend.position = "bottom",
    legend.title = element_blank()
  )

p2 
```

```{r}
ggsave("/home/kimberly.ledger/rockfish_mb/figures/rockfish_id_binary.png", plot = p1, dpi = 300, width = 12, height = 6)
ggsave("/home/kimberly.ledger/rockfish_mb/figures/rockfish_id_colors.png", plot = p2, dpi = 300, width = 12, height = 6)
```

```{r}
rockfish <- rockfish %>%
  filter(species_name != "Sebastolobus altivelis") %>%
  filter(species_name != "Sebastolobus macrochir") %>%  ## not going to worry about sebastolobus 
  filter(!sample_id %in% ablg_id$accession_id)
```

add species name to asv_table to help with troubleshooting
```{r}
asv_table_w_id <- asv_table %>%
  rename(sample_id = X) %>%
  left_join(ablg) %>%
  select(sample_id, species_name, common_name, everything())
```

```{r}
#asv_table_w_id %>%
#  filter(species_name %in% c("Sebastes diploproa"))
```

notes: 
* added S. diaconus to reference db and removed S. altivelis, and S. microchar from analysis due to availability of ref sequences
* I already knew that S. ciliatus and S. variablilis were not going to be possible to separate 
* S. diploproa tissues did not match any of the S. diploproa reference sequences... there were not a large number of reads for these tissues

```{r}
rockfish %>%
  filter(is.na(taxon)) %>%
  filter(prop_reads > 0.75)
```
